projects/js-packages/publicize-components/src/components/media-section/index.js

```diff
@@ -111,18 +146,36 @@ export default function MediaSection() {
 		[ attachedMedia[ 0 ] ]
 	);
 
-	const { mediaData, metaData } = getMediaDetails( mediaObject );
+	useEffect( () => {
+		try {
+			const setMediaDetails = async () => {
+				const mediaDetails = await getMediaDetails( mediaObject );
```

Comment:
Should we store the `mediaDetails` in the state and destructure it to get the `metaData` and `mediaData`?